meilisearch-http/src/snapshot.rs

```diff
@@ -0,0 +1,44 @@
+use std::path::{PathBuf, Path};
+use std::fs::{create_dir_all, File};
+use tar::Builder;
+use flate2::write::GzEncoder;
+use flate2::Compression;
+use std::thread;
+use std::time::{Duration};
+use tempfile::TempDir;
+use crate::error::Error;
+use crate::Data;
+
+
+pub fn create_snapshot(data: &Data, snapshot_path: &PathBuf) -> Result<(), Error> {
+    let tmp_dir = TempDir::new()?;
+
+    data.db.copy_and_compact_to_path(tmp_dir.path());
```

Comment:
Hello @MarinPostma @Kerollmops, here we have a `meilisearch_core::error::Error`,
should I reimplement `From<meilisearch_core::error::Error>` like we made [here](https://github.com/meilisearch/MeiliSearch/pull/829/files#diff-f6b32c1d41673385c68a91309c9b2dc0R239-R244) or is there a  better way to do it?

Reply:
As we don't plan to return it with an http, and we don't have anyhow yet, I suggest that we return a box dyn error, and just log it as you mentioned after.

Reply:
We'll turn it in an anyhow error later